nixos/modules/services/monitoring/scrutiny-collector.nix

```diff
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) mkEnableOption mkIf mkOption mkPackageOption types mdDoc;
+
+  cfg = config.services.scrutiny-collector;
+in
+{
+  options.services.scrutiny-collector = {
+    enable = mkEnableOption (mdDoc "scrutiny-collector");
+
+    package = mkPackageOption pkgs "scrutiny-collector" { };
+
+    config-path = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Specify the path to the devices file
+      '';
+    };
+
+    configuration = mkOption {
+      type = types.nullOr (pkgs.formats.yaml { }).type;
+      default = null;
+      description = lib.mdDoc ''
+        Specify the configuration for the Scrutiny collector in Nix.
+      '';
+    };
+
+    api-endpoint = mkOption {
+      type = types.str;
+      default = "http://localhost:8080";
+      description = mdDoc ''
+        The api server endpoint
+      '';
+    };
+
+    log-file = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Path to file for logging. Leave empty to use STDOUT
+      '';
+    };
+
+    debug = mkEnableOption "debug logging";
+
+    host-id = mkOption {
+      type = types.str;
+      default = config.networking.hostName;
+      defaultText = lib.literalExpression "config.networking.hostName";
+      description = mdDoc ''
+        Host identifier/label, used for grouping devices
+      '';
+    };
+
+    calendar = mkOption {
+      type = types.str;
+      default = "*-*-* 00:00:00";
+      description = mdDoc ''
+        Configured when to run the service systemd unit (DayOfWeek Year-Month-Day Hour:Minute:Second).
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "scrutiny-collector";
+      description = mdDoc ''
+        User under which the scrutiny collector service runs.
+      '';
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "scrutiny-collector";
+      description = mdDoc ''
+        Group under which the scrutiny collector service runs.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable
+    (
+      let
+        configFile =
+          if cfg.configuration != null then
+            (pkgs.writeTextFile {
+              name = "collector.yaml";
+              text = (lib.generators.toYAML { } cfg.configuration);
+            })
+          else null;
+        createUser = cfg.user == "scrutiny-collector";
+      in
+      {
+        environment.systemPackages = [ cfg.package ];
+
+        users.groups.${cfg.group} = lib.mkIf createUser { };
```

Comment:
does this not do nothing?

Reply:
The goal here was to only create the group if we're creating the user. Does this not do what I intended?

Reply:
was a genuine question on my part for what its worth, was just unable to tell at the time, if it works don't let me worry you about it, oops!

Reply:
Here's an example of creating a group with an empty attrset, and then adding a user to that group: https://superuser.com/a/1352988